the-collab-lab/tcl-62-smart-shopping-list
# Add item: Button, modal, and styling

## Description

When a user is on the List page, they will see a button to add a new item. If they have items on their list, the button will be located under their list. When they click on the Add Item button, they'll be presented with a modal to add their item and select when to purchase the item next. When the Add Item button is tapped, the user will see one of the following three Alerts: 
- Success: If the user has submitted a new item to their list successfully, they will see a green alert saying "Item successfully added to your list!"
- Failed - Empty field/s: If either of the fields are not filled out, the user will see a red alert saying "Item was NOT added because one or more fields were incomplete. Please fill out all fields."
- Failed - Duplicate item: If the user attempts to submit a duplicate item to the list, they'll see an orange alert with the message "You've already added this item, so we were not able to add it to your list." 

Note that currently, the font sizes are a bit off across the app. As a team, we will need to decide on how to build in consistency across the website, maybe setting a global font size. 
 

## Related Issue
closes 18


## Acceptance criteria
- [x] Create and style Add Item button
- [x] Create and style Add Item modal 
- [x] Add styling for drop-down menu to select when to next purchase item

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
| ✓ | Add styling              |

## Updates

### Before

<img width="1137" alt="Screenshot 2023-10-03 at 4 23 15 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/63386589/7c87c81a-9548-4d9e-bf48-7225af2fdadd">


### After

Add Item button
<img width="704" alt="Screenshot 2023-10-03 at 6 34 26 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/63386589/7ba1aae1-a933-409e-959b-75c7707f24a3">

Add Item modal
<img width="704" alt="Screenshot 2023-10-05 at 12 18 10 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/63386589/a44a3e69-c03e-4638-9799-bde29c879ada">

Alert after successful submission
<img width="704" alt="Screenshot 2023-10-03 at 6 34 54 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/63386589/a5d3889a-ea2e-4423-8089-8a4e345fd09c">

Alert after failed submission
<img width="704" alt="Screenshot 2023-10-03 at 6 35 04 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/63386589/8d2b75b1-e258-4c78-a67b-bfd1299be0ee">




## Testing Steps / QA Criteria

1. Open the main branch in your terminal and pull all remote changes with `git pull`
2. Navigate to this feature branch with command `git switch modal-add-item`
3. Run `npm start` to spin up the feature in the dev web environment.
4. In your browser, navigate to the List page of the dev website.
5. If you need to create a list, go ahead and do so. 
6. On the List page, tap the Add Item button, which will show you a modal. Both fields should be outlined in red indicating that it needs to be updated. Add a new item and select a time frame from the drop-down menu. The red outlines should no longer show. Click the Add Item button.
7. The modal should automatically close, and you should see a green alert saying "Item successfully added to your list!"
8. Go to the Firestore database to confirm your item was added to your list.
9. Navigate back to the List page and click the Add Item button. This time, leave either or both of the fields untouched and tap the Add Item button. 
10. The modal should automatically close, and you should see a red alert saying "Item was NOT added because one or more fields were incomplete. Please fill out all fields."
11. Navigate back to the List page and click the Add Item button. Now this time, enter a duplicated item, select a timeframe from the drop-down menu, and tap the Add Item button. 
12. The modal should automatically close, and you should see an orange alert with the message "You've already added this item, so we were not able to add it to your list."
